pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -160,8 +160,12 @@ func (r *ReconcileUpgradeConfig) Reconcile(request reconcile.Request) (reconcile
 			metricsClient.UpdateMetricValidationFailed(instance.Name)
 			return reconcile.Result{}, nil
 		}
-		reqLogger.Info("UpgradeConfig validated.")
 		metricsClient.UpdateMetricValidationSucceeded(instance.Name)
```

Comment:
only tiny nitpick, should we move this below the proceed block? otherwise we report metrics for no-ops.

Reply:
Funny you bring that up - I was thinking about the same when I made the change, and my eventual thinking was that - let's say a borked UpgradeConfig arrives that sets the failed metric and triggers an alert. If it's moved below the proceed block, our only way _out_ of that alerting state is to go forth and upgrade. I envisioned the scenario where maybe there's a bad UpgradeConfig pushed out (let's say, due to a missing edge), and we decide we need to backout and not upgrade at all.. we could push an UpgradeConfig with the cluster's current version, and reset the 'failed' metric without actually pursuing an upgrade.

I'm presenting a pretty convoluted scenario, but that was the motivation for shifting it up. I'm fine either way, your point on not reporting metrics for no-ops is sound. What do you think?

Reply:
> our only way out of that alerting state is to go forth and upgrade.
should a validated clear config clear the alert and not require a metric to infer validation?